### PR TITLE
fix/reset-password

### PR DESCRIPTION
**Pull request by SergKukh**
- remove email does not exist error
- added link valdation before page will be loaded
- refactored PasswordReset & PasswordResetRequest pages (moved logic into custom hooks)

image

image

image

### CODE REVIEW

**Review by pavlo-chernykh — APPROVED** [review]

File: apps/freelance/src/pages/PasswordReset/index.tsx [hunk 1]
```diff
@@ -1,100 +1,102 @@
...
-                  new Error(t('resetPassword.validation.passwordNotMatch')),
-                );
-              },
-            }),
-          ]}
-          hasFeedback
-        >
-          <StyledPasswordInput
-            placeholder={t('resetPassword.placeholder.password')}
+      {isLinkValid === true && (
```
Comment (lyudad) [inline apps/freelance/src/pages/PasswordReset/index.tsx:30]: `{isLinkValid &&}` should work

File: apps/freelance/src/pages/PasswordResetRequest/userPasswordResetRequest.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+interface UsePasswordResetRequestResponse {
+  handleSubmit: UseFormHandleSubmit<Inputs>;
+  onSubmit: (data: Inputs) => SubmitHandler<Inputs>;
+  control: Control<Inputs>;
+  email: string;
+  isLoading: boolean;
+  isError: boolean;
+  error: UserError | SerializedError | undefined;
```
Comment (lyudad) [inline apps/freelance/src/pages/PasswordResetRequest/userPasswordResetRequest.ts:23]: `error?: UserError | SerializedError;` should work

File: apps/freelance/src/pages/PasswordReset/index.tsx (hunk 1 above)
Comment (SergKukh) [reply apps/freelance/src/pages/PasswordReset/index.tsx:30]: @lyudad fixed

File: apps/freelance/src/pages/PasswordResetRequest/userPasswordResetRequest.ts (hunk 2 above)
Comment (SergKukh) [reply apps/freelance/src/pages/PasswordResetRequest/userPasswordResetRequest.ts:23]: @lyudad fixed

**Pull request merged**